graphdatascience/query_runner/neo4j_query_runner.py

```diff
@@ -124,10 +133,19 @@ def run_cypher(
             else:
                 self._last_bookmarks = session.last_bookmarks()
 
-            notifications = result.consume().notifications
-            if notifications:
-                for notification in notifications:
-                    self._forward_cypher_warnings(notification)
+            if (
+                Neo4jQueryRunner._NEO4J_DRIVER_VERSION < ServerVersion(5, 21, 0)
+                or result._warn_notification_severity != "WARNING"
+            ):
+                notifications = result.consume().notifications
+                if notifications:
+                    for notification in notifications:
+                        self._forward_cypher_warnings(notification)
+            elif Neo4jQueryRunner._NEO4J_DRIVER_VERSION > ServerVersion(5, 21, 0):
```

Comment:
this should be >=, no?

```suggestion
            elif Neo4jQueryRunner._NEO4J_DRIVER_VERSION >= ServerVersion(5, 21, 0):
```